### PR TITLE
Show celebration dialog after a user creates their first section

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Hooks up the celebration dialog! This only works on the new flow -- I would've had to do extra work to make it work in both flows.

This will also conflict with Hannah's PR. I'll update this one after her PR is merged.

Excuse my computer's slowness:
https://user-images.githubusercontent.com/46464143/234971487-fff1c7be-7252-41d0-a782-f8d74890985e.mov

### CODE REVIEW

File: apps/test/unit/templates/sectionsRefresh/SectionsSetUpContainerTest.jsx [hunk 1]
```diff
@@ -107,10 +110,47 @@ describe('SectionsSetUpContainer', () => {
...
     expect(fetchSpy).to.have.been.called.once;
 
+    await new Promise(resolve => setTimeout(resolve, 0));
+    expect(navigateToHrefSpy).to.have.been.called.once;
+    expect(navigateToHrefSpy.getCall(0).args[0]).to.include('/home');
+
+    sinon.restore();
+  });
+
+  it('appends showSectionCreationDialog to url if isUsersFirstSection is true', async () => {
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/sectionsRefresh/SectionsSetUpContainerTest.jsx:116]: This worked locally before I merged staging in. I kept these commented out as similar tests were commented out

File: dashboard/app/controllers/sections_controller.rb [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
   before_action :load_section_by_code, only: [:log_in, :show]
+  authorize_resource :section, only: [:new]
 
   def new
-    return head :forbidden unless current_user&.admin
+    return head :forbidden unless current_user.admin
 
     redirect_to '/home' unless params[:loginType] && params[:participantType]
+
+    @is_users_first_section = current_user.sections.empty?
```
Comment (hannahbergam) [inline dashboard/app/controllers/sections_controller.rb:11]: This is really nice

**Review by hannahbergam — COMMENTED** [review]
I love love love watching that video. For a teacher importing an oauth section- I know the is_users_first_section param will be correct, we just won't have the redirect set up right?

**Comment by bethanyaconnor** [comment]
> For a teacher importing an oauth section- I know the is_users_first_section param will be correct, we just won't have the redirect set up right?

ooo good point -- I think this wouldn't work for the first oauth section (it would be saved before directing to the edit page I think). Not sure how big of a deal this is?

**Comment by hannahbergam** [comment]
> > For a teacher importing an oauth section- I know the is_users_first_section param will be correct, we just won't have the redirect set up right?
> 
> ooo good point -- I think this wouldn't work for the first oauth section (it would be saved before directing to the edit page I think). Not sure how big of a deal this is?

I think probably not a big deal? In an ideal world (maybe future future work?) we would have a way to keep track of of new oauth sections, but we've been ignoring that case in prod for a long time anyway😂 (i.e. sending oauth to an 'edit section' screen even though it's brand new)

File: apps/test/unit/templates/sectionsRefresh/SectionsSetUpContainerTest.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/test/unit/templates/sectionsRefresh/SectionsSetUpContainerTest.jsx:116]: My [pr here](https://github.com/code-dot-org/code-dot-org/pull/51676) will relaunch these tests!

**Review by hannahbergam — APPROVED** [review]
We chatted offline- this looks great! Thanks for handling the merge conflict. Can't wait to see it in action:)

**Pull request merged**